batched/dense/unit_test/Test_Batched_SerialPttrf.hpp

```diff
@@ -0,0 +1,401 @@
+                                                             _beta, cc);
+  }
+
+  inline void run() {
+    using value_type = typename AViewType::non_const_value_type;
+    std::string name_region("KokkosBatched::Test::SerialPttrf");
+    const std::string name_value_type = Test::value_type_name<value_type>();
+    std::string name                  = name_region + name_value_type;
+    Kokkos::RangePolicy<execution_space> policy(0, _a.extent(0));
+    Kokkos::parallel_for(name.c_str(), policy, *this);
+  }
+};
+
+template <typename DeviceType, typename ScalarType, typename LayoutType,
+          typename AlgoTagType>
+/// \brief Implementation details of batched pttrf test for random matrix
+///
+/// \param N [in] Batch size of matrix A
+/// \param BlkSize [in] Block size of matrix A
+void impl_test_batched_pttrf(const int N, const int BlkSize) {
+  using real_type      = typename real_type<ScalarType>::type;
+  using RealView2DType = Kokkos::View<real_type **, LayoutType, DeviceType>;
+  using View2DType     = Kokkos::View<ScalarType **, LayoutType, DeviceType>;
+  using View3DType     = Kokkos::View<ScalarType ***, LayoutType, DeviceType>;
+
+  View3DType A("A", N, BlkSize, BlkSize),
+      A_reconst("A_reconst", N, BlkSize, BlkSize);
+  View3DType EL("EL", N, BlkSize, BlkSize), EU("EU", N, BlkSize, BlkSize),
+      D("D", N, BlkSize, BlkSize), LD("LD", N, BlkSize, BlkSize),
+      L("L", N, BlkSize, BlkSize), I("I", N, BlkSize, BlkSize);
+  RealView2DType d("d", N, BlkSize),
+      ones("ones", N, BlkSize);  // Diagonal and sub diagnoal components
+  View2DType e_upper("e_upper", N, BlkSize - 1),
+      e_lower("e_lower", N,
+              BlkSize - 1);  // upper and lower diagnoal components
+
+  using execution_space = typename DeviceType::execution_space;
+  Kokkos::Random_XorShift64_Pool<execution_space> rand_pool(13718);
+  real_type realRandStart = 0.0, realRandEnd = 1.0;
+  ScalarType randStart = 0.0, randEnd = 1.0;
+
+  KokkosKernels::Impl::getRandomBounds(1.0, realRandStart, realRandEnd);
+  KokkosKernels::Impl::getRandomBounds(1.0, randStart, randEnd);
+  Kokkos::fill_random(d, rand_pool, realRandStart, realRandEnd);
+  Kokkos::fill_random(e_upper, rand_pool, randStart, randEnd);
+
+  auto h_d       = Kokkos::create_mirror_view(d);
+  auto h_e_upper = Kokkos::create_mirror_view(e_upper);
+  auto h_e_lower = Kokkos::create_mirror_view(e_lower);
+  auto h_ones    = Kokkos::create_mirror_view(ones);
+
+  for (int ib = 0; ib < N; ib++) {
+    for (int i = 0; i < BlkSize; i++) {
+      h_d(ib, i) += static_cast<real_type>(
+          BlkSize);  // Add BlkSize to ensure positive definiteness
+      h_ones(ib, i) = 1;
+    }
+
+    for (int i = 0; i < BlkSize - 1; i++) {
+      // FIXME: We cannot use complex conjugate for real type
+      h_e_upper(ib, i) =
+          Kokkos::ArithTraits<ScalarType>::real(h_e_upper(ib, i));
+
+      // Fill the lower diagnocal with conjuate of the upper diagnoal
+      h_e_lower(ib, i) =
```

Comment:
What's the point if you removed the complex part previously? Just copy values of `e_upper` to `e_lower` using `deep_copy` or actually use complex values.